lib/csu/aarch64/crt1_c.c

```diff
@@ -77,21 +77,16 @@ __start(int argc, char *argv[], char *env[], void (*cleanup)(void))
 	 * would initialize them if they did.
 	 */
 	if (&_DYNAMIC == NULL) {
-		const Elf_Phdr * __capability phdr_cap = NULL;
 		const Elf_Auxinfo *auxp;
 		char **strp;
 		void *phdr = NULL;
 		long phnum = 0;
 
 		strp = env;
-		while (*strp == NULL)
-			strp++;
-		auxp = (Elf_Auxinfo *)(strp + 1);
-
-		/*
-		 * Digest the auxiliary vector to get information
-		 * needed to init globals.
-		 */
+		while (*strp++ != NULL)
+			;
```

Comment:
Not particularly keen on having `;` loop bodies.
Maybe
```suggestion
		while (*strp != NULL)
			strp++;
```
or
```suggestion
		while (*strp++ != NULL) {}
```
or
```suggestion
		while (*strp++ != NULL) {
		}
```

Reply:
The first suggestion is exactly what it was before and would thus need the `+ 1` restored on the line below. I chose this form because (other than NULL vs 0) it is precisely how init_aux_vector_once styles it (see the commit message).